libs/k8s-client/tests/BaseApi/PodWithLogStreamTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\K8sClient\Tests\PodWithLogStream;
+
+use GuzzleHttp\Handler\CurlHandler;
+use GuzzleHttp\HandlerStack;
+use Keboola\K8sClient\BaseApi\PodWithLogStream;
+use KubernetesRuntime\Client;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+use ReflectionClass;
+
+class PodWithLogStreamTest extends TestCase
+{
+    public function testReadLogReturnsBodyOnSuccess(): void
+    {
+        $namespace = 'default';
+        $podName = 'test-pod';
+        $queries = ['foo' => 'bar'];
+
+        $responseMock = $this->createMock(ResponseInterface::class);
+        $responseMock->method('getStatusCode')->willReturn(200);
+        $responseBody = $this->createMock(StreamInterface::class);
+        $responseMock->method('getBody')->willReturn($responseBody);
+
+        $clientMock = $this->createMock(Client::class);
+
+        $clientMock->expects($this->once())
+            ->method('request')
+            ->with(
+                $this->equalTo('get'),
+                $this->equalTo(sprintf('/api/v1/namespaces/%s/pods/%s/log', $namespace, $podName)),
+                $this->callback(function ($options) use ($queries) {
+                    $this->assertArrayHasKey('query', $options);
+                    $this->assertEquals($queries, $options['query']);
+
+                    // Check that 'handler' is set and is an instance of HandlerStack
+                    $this->assertArrayHasKey('handler', $options);
+                    $this->assertInstanceOf(HandlerStack::class, $options['handler']);
+
+                    $handlerStack = $options['handler'];
+                    $reflection = new ReflectionClass($handlerStack);
+                    $property = $reflection->getProperty('handler');
+                    $property->setAccessible(true);
+                    $handler = $property->getValue($handlerStack);
```

Comment:
Na to mame https://github.com/keboola/sandboxes-service/blob/cbc4983a7e44df05708dd6947f35b432e9cb8af5/tests/ReflectionPropertyAccessTestCase.php#L21